tests/models/test_dagrun.py

```diff
@@ -229,7 +229,7 @@ def test_dagrun_deadlock(self):
                                start_date=now)
 
         ti_op1 = dr.get_task_instance(task_id=op1.task_id)
-        ti_op1.set_state(state=State.SUCCESS, session=session)
+        ti_op1.set_state(state=State.FAILED, session=session)
```

Comment:
Why did you need to change this test? It doesn't look like it should need any changes

Reply:
The trigger rule was defined to be one_failed which means (surprisingly) that one task has to fail. what happened before the change is: It resulted in deadlock right away and failed the dag which supposed to be in 'running' state. 

Why did it work before the refactor?
probably because it used another session and took the data from the db which said that the task is 'running' or 'none' 

now I might not understand this test fully so please correct me if you think it is not the case.

Reply:
Let me double check the rest of this test in detail then. You may be right, I just wanted to double check as this affects right in to the core of Airflow :)

Reply:
(This test is so badly named -- that is not a "deadlock".)

Reply:
Okay, I've cone back to the original commit where we introduced the concept of deadlocks: 24a7072990fb232a6fe23c9eb3fdb72ba0695c96 and it says this:

```
3. Deadlock — A dag run is deadlocked when no action is possible.
This is determined by the presence of unfinished tasks without met
dependencies. However, care must be taken when depends_on_past=True
because individual dag runs could *look* like they are deadlocked
when they are actually just waiting for earlier runs to finish.
```

(not what I'd traditionally call a deadlock, but okay).

So I think this is _wrong_. Since the task is in success state, but the trigger rule is one_failed it should cause the "deadlock" (or also known as terminal state): when ti_op1 is in success state ti_op2 won't ever be able to run, so this should cause it to fail

Reply:
Wait. What? I..... am not sure of anything anymore

Reply:
You are right, But it means the dr should have been 'failed' instead of 'running'. I will change it the task to 'success' and the state of the dagrun to 'failed' and then it will be as intended.